build_docs

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python2
+
+# Build the docs with docker!
+#
+# Step 1 is to build a docker image based on the asciidoctor image.
+# Step 2 is to translate the arguments that build_docs.pl supports into
+# a list of arguments to be passed to start the docker container and a
+# list of arguments to be passed to the build_docs.pl process that is
+# started in the docker container.
+# Step 3 is to start the docker container. We start it in such a way
+# that is *should* remove itself when it is done.
+#
+# I'm aware that this is a fairly non-standard way to use docker but
+# for the most part all we want docker for is to get us into a consistent
+# environment it'll do the trick. At least for now.
+
+import logging
+from os import environ, getgid, getuid
+from os.path import basename, dirname, exists, isdir, join, realpath
+import subprocess
+from sys import platform, stdout
+import time
+
+DOCKER_BUILD_QUIET_TIME = 3 # seconds
+
+DIR = dirname(realpath(__file__))
+logger = logging.getLogger('build_docs')
+
+def build_docker_image():
+    logger.info('Building the docker image that will build the docs. Expect '
+        + 'this to take somewhere between a hundred milliseconds and '
+        + 'five minutes.')
+    with open(join(DIR, 'Dockerfile')) as dockerfile:
+        # We attempt to spool up the output from docker build so we can hide it
+        # if the command is successful *and* runs quickly. If it takes longer
+        # than the DOCKER_BUILD_QUIET_TIME then we log all of the output. I
+        # know this is a little sneaky, but folks run this from the command
+        # line so frequently that they'll appreciate cleaner output. I hope.
+        cmd = ["docker", "image", "build", "-t", "elastic/docs_build", "-"]
+        build = subprocess.Popen(
+            cmd,
+            stdin = dockerfile,
+            stdout = subprocess.PIPE,
+            stderr = subprocess.STDOUT)
+        docker_logger = logging.getLogger('docker build')
+        start_logging_at = time.time() + DOCKER_BUILD_QUIET_TIME
+        acc = []
+        while True:
+            line = build.stdout.readline()
+            if line != '':
+                line = line.rstrip()
```

Comment:
Any reason not to simply iterate `for line in build.stdout`?

Reply:
I recall that I did this because of something to do with buffering, but I'll recheck.